futures-core/src/future/mod.rs

```diff
@@ -8,6 +8,19 @@ pub use core::future::Future;
 mod future_obj;
 pub use self::future_obj::{FutureObj, LocalFutureObj, UnsafeFutureObj};
 
+/// A `Future` or `TryFuture` which tracks whether or not the underlying future
+/// should no longer be polled.
+///
+/// `has_terminated` will return `true` if a future should no longer be polled.
+/// Usually, this state occurs after `poll` (or `try_poll`) returned
+/// `Poll::Ready`. However, `has_terminated` may also return `false` if a future
+/// has become inactive and can no longer make progress and should be ignored
+/// or dropped rather than being `poll`ed again.
+pub trait FusedFuture {
+    /// Returns `true` if the underlying future should no longer be polled.
+    fn has_terminated(&self) -> bool;
```

Comment:
I believe `is_` is more idiomatic than `has_`

There are a number of `is_` fns in the repo already, but I did not find a fn prefixed with `has_` (this would be the first.

Reply:
Sure, done.